e2e-tests/tfm/cypress/e2e/journeys/case-amendments/facility-end-date/feature-flag-enabled/gef-deal-add-bank-review-date.spec.js

```diff
@@ -0,0 +1,149 @@
+import { MOCK_FACILITY_ONE } from '../../../../../fixtures/mock-gef-facilities';
+import { MOCK_APPLICATION_AIN } from '../../../../../fixtures/mock-gef-deals';
+import { DEAL_TYPE } from '../../../../../../../gef/cypress/fixtures/constants';
+import amendmentsPage from '../../../../pages/amendments/amendmentsPage';
+import dateConstants from '../../../../../../../e2e-fixtures/dateConstants';
+
+if (Cypress.env('FF_TFM_FACILITY_END_DATE_ENABLED') === 'true') {
+  context('Amendments - GEF deal add bank review date', () => {
+    let dealId;
+    let dealFacilities;
+
+    const MOCK_GEF_FACILITY = {
+      ...MOCK_FACILITY_ONE,
+      hasBeenIssued: true,
+      isUsingFacilityEndDate: undefined,
+      facilityEndDate: undefined,
+      bankReviewDate: undefined,
+    };
+
+    before(() => {
+      // inserts a gef deal
+      cy.insertOneGefDeal(MOCK_APPLICATION_AIN, BANK1_MAKER1).then((insertedDeal) => {
+        dealId = insertedDeal._id;
+        // updates a gef deal so has relevant fields
+        cy.updateGefDeal(dealId, MOCK_APPLICATION_AIN, BANK1_MAKER1);
+
+        cy.createGefFacilities(dealId, [MOCK_GEF_FACILITY], BANK1_MAKER1).then((createdFacilities) => {
+          dealFacilities = createdFacilities.details;
+        });
+
+        cy.submitDeal(dealId, DEAL_TYPE.GEF, T1_USER_1);
+      });
+    });
+
+    beforeEach(() => {
+      cy.login(PIM_USER_1);
+      cy.visit(relative(`/case/${dealId}/facility/${dealFacilities._id}`));
+    });
+
+    after(() => {
+      cy.deleteDeals(dealId, ADMIN);
+    });
+
+    it('should display the default blank facility end date fields in the Details tab', () => {
+      facilityPage.facilityIsUsingFacilityEndDate().should('have.text', '-');
+      facilityPage.facilityFacilityEndDate().should('have.text', '-');
+      facilityPage.facilityBankReviewDate().should('not.exist');
+    });
+
+    it('should navigate to the bank review date page after selecting "No" to "Has the bank provided a facility end date"', () => {
+      amendmentsPage.navigateToIsUsingFacilityEndDatePage(true, false);
+      amendmentsPage.isUsingFacilityEndDateYes().should('be.not.checked');
+      amendmentsPage.isUsingFacilityEndDateNo().should('be.not.checked');
+
+      amendmentsPage.continueAmendment().click();
+      amendmentsPage.errorSummary().contains('Select if the bank has provided an end date for this facility');
+
+      amendmentsPage.isUsingFacilityEndDateNo().click();
+      amendmentsPage.continueAmendment().click();
+
+      cy.url().should('contain', 'bank-review-date');
+      amendmentsPage.amendmentCurrentBankReviewDate().should('have.text', 'Not provided');
+      amendmentsPage.amendmentBankReviewDateDetails().should('exist');
+    });
+
+    it('should return expected errors when entering an incorrect bank review date', () => {
+      amendmentsPage.navigateToIsUsingFacilityEndDatePage(false, false);
```

Comment:
This function might be a bit clearer if it took an object of optional parameters which default to false?

Reply:
Good idea, have changed